shared/middleware/channels_api.rb

```diff
@@ -47,19 +47,34 @@ class ChannelsApi < Sinatra::Base
   #
   # Create a channel.
   #
+  # Optional query string param: ?src=<src-channel-id> creates the channel as
+  # a copy of the given src channel.
+  #
   post '/v3/channels' do
     unsupported_media_type unless request.content_type.to_s.split(';').first == 'application/json'
     unsupported_media_type unless request.content_charset.to_s.downcase == 'utf-8'
 
-    begin
-      data = JSON.parse(request.body.read)
-    rescue JSON::ParserError
-      bad_request
+    src_channel = request.GET['src']
+    storage_app = StorageApps.new(storage_id('user'))
```

Comment:
Probably shouldn't block this particular change then, but may or may not be something we want to address in the future.